File: demo-android/app/src/androidTest/java/com/example/demo/e2e/data/TaskResponse.kt
package com.example.demo.e2e.data

import com.example.demo.data.model.Task
import com.example.demo.data.model.TaskPriority
import com.example.demo.data.model.TaskStatus

data class TaskResponse(
    val id: String,
    val title: String,
    val description: String,
    val status: TaskStatus,
    val priority: TaskPriority,
) {
    fun toTask(): Task = Task(
        id = id,
        title = title,
        description = description,
        status = status,
        priority = priority,
        createdDate = MOCK_CREATED_DATE,
        updatedDate = MOCK_UPDATED_DATE,
    )

    private companion object {
        const val MOCK_CREATED_DATE = "2024-01-01T10:00:00Z"
        const val MOCK_UPDATED_DATE = "2024-01-02T10:00:00Z"
    }
}

File: e2e/maestro/data/TaskResponse.ts
import { TaskPriority } from './TaskPriority';
import { TaskStatus } from './TaskStatus';

export type WireMockTask = {
  id: string;
  title: string;
  description: string;
  status: TaskStatus;
  priority: TaskPriority;
  createdDate: string;
  updatedDate: string;
};

export type TaskResponse = {
  id: string;
  title: string;
  description: string;
  status: TaskStatus;
  priority: TaskPriority;
};

const MOCK_CREATED_DATE = '2024-01-01T10:00:00Z';
const MOCK_UPDATED_DATE = '2024-01-02T10:00:00Z';

export function toWireMockTask(response: TaskResponse): WireMockTask {
  return {
    id: response.id,
    title: response.title,
    description: response.description,
    status: response.status,
    priority: response.priority,
    createdDate: MOCK_CREATED_DATE,
    updatedDate: MOCK_UPDATED_DATE,
  };
}
